nixos/modules/services/networking/jool.nix

```diff
@@ -0,0 +1,277 @@
+{ config, pkgs, lib, ... }:
+
+let
+  cfg = config.networking.jool;
+
+  jool = config.boot.kernelPackages.jool;
+  jool-cli = pkgs.jool-cli;
+
+  hardening = {
+    # Run as unprivileged user
+    User = "jool";
+    Group = "jool";
+    DynamicUser = true;
+
+    # Restrict filesystem to only read the jool module
+    TemporaryFileSystem = [ "/" ];
+    BindReadOnlyPaths = [
+      builtins.storeDir
+      "/run/current-system/kernel-modules"
```

Comment:
Shouldn't this be `/run/booted-system` ?

This will fail on system if you don't reboot to have the new kernel for the current-system, which is a bit of a shame IMHO.

Reply:
I'll have to test this, I'm not sure what happens after switching the kernel.

Reply:
I can't easiliy test switching to an incompatible kernel/kernel module, however /run/booted-system seems to work too, and it makes more sense.